File: VB/FocusRectHelper/ThickFocusRectHelper.vb
Imports DevExpress.Utils.Paint
Imports System
Imports System.Drawing
Imports System.Reflection

Namespace CellBorder.FocusRectHelper
    Friend NotInheritable Class ThickFocusRectHelper

        Private Sub New()
        End Sub

        Public Shared Sub SetupCustomPainter()
            XPaint.CreateCustomPainter(New MyXPaint())
        End Sub

        Public Shared Sub DrawThickFocusRectangle(ByVal g As Graphics, ByVal r As Rectangle)
            Dim hb As Brush = Brushes.Black
            g.FillRectangle(hb, New Rectangle(r.X, r.Y, 2, r.Height - 2)) ' left
            g.FillRectangle(hb, New Rectangle(r.X, r.Y, r.Width - 2, 2)) ' top
            g.FillRectangle(hb, New Rectangle(r.Right - 2, r.Y, 2, r.Height - 2)) ' right
            g.FillRectangle(hb, New Rectangle(r.X, r.Bottom - 2, r.Width, 2)) ' bottom
        End Sub
    End Class
    Public Class MyXPaint
        Inherits XPaint

        Public Overrides Sub DrawFocusRectangle(ByVal g As Graphics, ByVal r As Rectangle, ByVal foreColor As Color, ByVal backColor As Color)
            If Not CanDraw(r) Then
                Return
            End If
            ThickFocusRectHelper.DrawThickFocusRectangle(g, r)
        End Sub
    End Class
    Public Class MyXPaintMixed
        Inherits XPaintMixed

        Public Overrides Sub DrawFocusRectangle(ByVal g As Graphics, ByVal r As Rectangle, ByVal foreColor As Color, ByVal backColor As Color)
            If Not CanDraw(r) Then
                Return
            End If
            ThickFocusRectHelper.DrawThickFocusRectangle(g, r)
        End Sub
    End Class
End Namespace

File: CS/FocusRectHelper/ThickFocusRectHelper.cs
using DevExpress.Utils.Paint;
using System;
using System.Drawing;
using System.Reflection;

namespace CellBorder.FocusRectHelper {
    static class ThickFocusRectHelper {
        public static void SetupCustomPainter() {
            XPaint.CreateCustomPainter(new MyXPaint());
        }

        public static void DrawThickFocusRectangle(Graphics g, Rectangle r) {
            Brush hb = Brushes.Black;
            g.FillRectangle(hb, new Rectangle(r.X, r.Y, 2, r.Height - 2)); // left
            g.FillRectangle(hb, new Rectangle(r.X, r.Y, r.Width - 2, 2));  // top
            g.FillRectangle(hb, new Rectangle(r.Right - 2, r.Y, 2, r.Height - 2)); // right
            g.FillRectangle(hb, new Rectangle(r.X, r.Bottom - 2, r.Width, 2)); // bottom
        }
    }
    public class MyXPaint : XPaint {
        public override void DrawFocusRectangle(Graphics g, Rectangle r, Color foreColor, Color backColor) {
            if (!CanDraw(r)) return;
            ThickFocusRectHelper.DrawThickFocusRectangle(g, r);
        }
    }
    public class MyXPaintMixed : XPaintMixed {
        public override void DrawFocusRectangle(Graphics g, Rectangle r, Color foreColor, Color backColor) {
            if (!CanDraw(r)) return;
            ThickFocusRectHelper.DrawThickFocusRectangle(g, r);
        }
    }
}
